CONTRIBUTING.md

```diff
@@ -1,106 +1,157 @@
+1. Open a pull request, linking to [GitHub search results][search-example] showing in-the-wild usage.
+   Please state clearly the license covering the code in the samples.
+   Link directly to the original source if possible.
 
 In addition, if your new language defines an extension that's already listed in [`languages.yml`][languages] (such as `.foo`) then sometimes a few more steps will need to be taken:
 
 1. Make sure that example `.foo` files are present in the [samples directory][samples] for each language that uses `.foo`.
-1. Test the performance of the Bayesian classifier with a relatively large number (1000s) of sample `.foo` files. (ping **@lildude** to help with this) to ensure we're not misclassifying files.
-1. If the Bayesian classifier does a bad job with the sample `.foo` files then a [heuristic](https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb) may need to be written to help.
+1. Test the performance of the Bayesian classifier with a relatively large number (1000s) of sample `.foo` files (ping **@lildude** to help with this).
+   This ensures we're not misclassifying files.
+1. If the Bayesian classifier does a bad job with the sample `.foo` files, then a [heuristic][] may need to be written to help.
 
 Remember, the goal here is to try and avoid false positives!
 
 
 ## Fixing a misclassified language
 
-Most languages are detected by their file extension defined in [`languages.yml`][languages].  For disambiguating between files with common extensions, Linguist applies some [heuristics](/lib/linguist/heuristics.rb) and a [statistical classifier](lib/linguist/classifier.rb). This process can help differentiate between, for example, `.h` files which could be either C, C++, or Obj-C.
+Most languages are detected by their file extension defined in [`languages.yml`][languages].
+For disambiguating between files with common extensions, Linguist applies some [heuristics](/lib/linguist/heuristics.rb) and a [statistical classifier](lib/linguist/classifier.rb).
+This process can help differentiate between, for example, `.h` files which could be either C, C++, or Obj-C.
 
-Misclassifications can often be solved by either adding a new filename or extension for the language or adding more [samples][samples] to make the classifier smarter.
+Misclassifications can often be solved by either adding a new filename or extension for the language or adding more [samples][] to make the classifier smarter.
 
 
 ## Fixing syntax highlighting
 
-Syntax highlighting in GitHub is performed using TextMate-compatible grammars. These are the same grammars that TextMate, Sublime Text and Atom use. Every language in [`languages.yml`][languages] is mapped to its corresponding TextMate `scopeName`. This scope name will be used when picking up a grammar for highlighting.
+Syntax highlighting in GitHub is performed using TextMate-compatible grammars.
+These are the same grammars that TextMate, Sublime Text and Atom use.
```

Comment:
We should add a serial comma here if we want to be consistent with the rest of the document.
```suggestion
These are the same grammars that TextMate, Sublime Text, and Atom use.
```

Reply:
Fixed + updated with a slightly-better wording. 👍